clang/lib/Lex/DependencyDirectivesScanner.cpp

```diff
@@ -503,6 +503,10 @@ bool Scanner::lexModuleDirectiveBody(DirectiveKind Kind, const char *&First,
           diag::err_dep_source_scanner_missing_semi_after_at_import);
     if (Tok.is(tok::semi))
       break;
+    if (Tok.is(tok::hash) || Tok.is(tok::at))
+      return reportError(
```

Comment:
IMHO there's no benefit in emitting an error during minimization, unless there's a reason that the frontend cannot emit a proper diagnostic during compilation.
The scanner's primary focus should be on identifying dependencies; invalid code can be reported by the frontend.

Reply:
The `#import` on its own line should be accepted (show up in minimization) I think?

Reply:
I'm not really sure how to do that without introducing some backtracking. AFAICT, as we consume each token, the `First` keeps moving forward.

Reply:
I think this should work fine for the test case:
```
@@ -496,7 +496,12 @@ bool Scanner::lexModuleDirectiveBody(DirectiveKind Kind, const char *&First,
                                      const char *const End) {
   const char *DirectiveLoc = Input.data() + CurDirToks.front().Offset;
   for (;;) {
+    const char *PriorFirst = First;
     const dependency_directives_scan::Token &Tok = lexToken(First, End);
+    if (Tok.is(tok::hash) && Tok.Flags & clang::Token::StartOfLine) {
+      First = PriorFirst;
+      return false;
+    }
     if (Tok.is(tok::eof))
       return reportError(
           DirectiveLoc,
```

Reply:
Thank you for taking a look! Applied your suggested patch, with minor tweaks.